### PR TITLE
KCI-9 📦 Upgrade ArgoCD from 2.1.7 to 2.5.6

### PR DESCRIPTION
**Pull request by deifyed**
## Description

- Moves generated files to `build/`
- Upgrades values template and chart version to 2.5.6
- Adds upgrade instructions in UPGRADE.md

## Motivation and Context

[KCI-9](https://github.com/oslokommune/kjoremiljo-confidential-issues/issues/9)

## How to prove the effect of this PR?

1. Open the ArgoCD UI at argocd.<your cluster>.oslo.systems and take note of the current status
2. Follow the instructions in UPGRADE.md
3. Verify that the ArgoCD UI is still available and has the noted status. Things you can check:
    - `kubectl get pods --namespace argocd` and verify that the pods are running and healthy
    - ArgoCD UI -> Repositories -> Verify that the repository connection is successful

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: gists/helm-component-extraction/argocd/UPGRADE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Upgrade ArgoCD
+
+## From 2.1.7 to 2.5.6
+
+### Back up existing applications
+
+```shell
+make applications-snapshot
```
Comment (ivaruf) [inline gists/helm-component-extraction/argocd/UPGRADE.md:8]: kubectl -n argocd get applications -o yaml \
	| yq '.items[] |= del(.status)' \
	| yq '.items[] |= del(.metadata.creationTimestamp)' \
	| yq '.items[] |= del(.metadata.generation)' \
	| yq '.items[] |= del(.metadata.resourceVersion)' \
	| yq '.items[] |= del(.metadata.uid)' > build/applications-snapshot.yaml
/bin/sh: 6: cannot create build/applications-snapshot.yaml: Directory nonexistent
make: *** [Makefile:190: applications-snapshot] Error 2

File: gists/helm-component-extraction/argocd/UPGRADE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+### Upgrade ArgoCD
+
+```shell
+# Prepare upgraded values
+make configure-helm-chart
+# Install upgraded chart with values
+make install-helm-chart
```
Comment (ivaruf) [inline gists/helm-component-extraction/argocd/UPGRADE.md:17]: make install-helm-chart
kubectl apply -f build/namespace.yaml
Warning: resource namespaces/argocd is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
namespace/argocd configured
kubectl apply -f build/argocd-secret.yaml
Warning: resource externalsecrets/argocd-secret is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
externalsecret.kubernetes-client.io/argocd-secret configured
helm repo add argo https://argoproj.github.io/argo-helm
"argo" already exists with the same configuration, skipping
helm repo update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "autoscaler" chart repository
...Successfully got an update from the "eks" chart repository
...Successfully got an update from the "aws-ebs-csi-driver" chart repository
...Successfully got an update from the "external-secrets" chart repository
...Successfully got an update from the "grafana" chart repository
...Successfully got an update from the "argo" chart repository
...Successfully got an update from the "prometheus-community" chart repository
Update Complete. ⎈Happy Helming!⎈
helm upgrade --install --namespace argocd argocd argo/argo-cd --version 5.17.1 --values build/values.yaml 
Error: UPGRADE FAILED: chart requires kubeVersion: >=1.22.0-0 which is incompatible with Kubernetes v1.21.14-eks-fb459a0
make: *** [Makefile:134: install-helm-chart] Error 1

File: gists/helm-component-extraction/argocd/UPGRADE.md (hunk 1 above)
Comment (deifyed) [reply gists/helm-component-extraction/argocd/UPGRADE.md:8]: Fixed

File: gists/helm-component-extraction/argocd/UPGRADE.md (hunk 2 above)
Comment (deifyed) [reply gists/helm-component-extraction/argocd/UPGRADE.md:17]: Interesting. Nice find

**Review by yngvark — APPROVED** [review]
I tested the INSTALL.md, since I didn't enable ArgoCD in my cluster-dev.yaml, so I could not upgrade. Works fine :+1:

**Comment by yngvark** [comment]
Tested on 1.21 btw

**Pull request merged**